### ISSUE TITLE
database is locked

### ISSUE THREAD

**Issue by abdulaziz-git** [adamhnat/vaultwarden-azure#1]
```
2023-02-13T08:59:29.18295  Connecting to the container 'vaultwarden'...
2023-02-13T08:59:29.20166  Successfully Connected to container: 'vaultwarden' [Revision: 'vaultwarden--viiytql', Replica: 'vaultwarden--viiytql-5d9676654f-thnw4']
2023-02-13T08:59:16.416097533Z /--------------------------------------------------------------------\
2023-02-13T08:59:16.416116327Z |                        Starting Vaultwarden                        |
2023-02-13T08:59:16.416120379Z |                           Version 1.27.0                           |
2023-02-13T08:59:16.416123040Z |--------------------------------------------------------------------|
2023-02-13T08:59:16.416125529Z | This is an *unofficial* Bitwarden implementation, DO NOT use the   |
2023-02-13T08:59:16.416128117Z | official channels to report bugs/features, regardless of client.   |
2023-02-13T08:59:16.416130587Z | Send usage/configuration questions or feature requests to:         |
2023-02-13T08:59:16.416133141Z |   https://vaultwarden.discourse.group/                             |
2023-02-13T08:59:16.416135620Z | Report suspected bugs/issues in the software itself at:            |
2023-02-13T08:59:16.416138069Z |   https://github.com/dani-garcia/vaultwarden/issues/new            |
2023-02-13T08:59:16.416140587Z \--------------------------------------------------------------------/
2023-02-13T08:59:16.416143190Z 
2023-02-13T08:59:16.416156555Z [INFO] No .env file found.
2023-02-13T08:59:16.416159281Z 
2023-02-13T08:59:16.685575661Z [2023-02-13 08:59:16.685][start][INFO] Rocket has launched from http://0.0.0.0:80
2023-02-13T08:59:51.270596732Z [2023-02-13 08:59:51.270][request][INFO] GET /admin
2023-02-13T08:59:51.270642435Z [2023-02-13 08:59:51.270][_][WARN] Request guard `AdminToken` failed: "Unauthorized".
2023-02-13T08:59:51.270649394Z [2023-02-13 08:59:51.270][_][WARN] Responding with registered (admin_login) /admin 401 catcher.
2023-02-13T08:59:51.271388886Z [2023-02-13 08:59:51.270][response][INFO] (admin_page) GET /admin/ => 401 Unauthorized
2023-02-13T09:00:18.771248342Z [2023-02-13 09:00:18.771][request][INFO] POST /admin
2023-02-13T09:00:18.774456116Z [2023-02-13 09:00:18.774][response][INFO] (post_admin_login) POST /admin/ => 200 OK
2023-02-13T09:00:25.939387489Z [2023-02-13 09:00:25.939][request][INFO] GET /admin/users/overview
2023-02-13T09:00:25.946444656Z [2023-02-13 09:00:25.946][response][INFO] (users_overview) GET /admin/users/overview => 200 OK
2023-02-13T09:00:37.883617368Z [2023-02-13 09:00:37.883][request][INFO] POST /admin/invite/
2023-02-13T09:00:42.969183140Z [2023-02-13 09:00:42.969][error][ERROR] Error saving invitation.
2023-02-13T09:00:42.969220638Z [CAUSE] DatabaseError(
2023-02-13T09:00:42.969226634Z     Unknown,
2023-02-13T09:00:42.969231258Z     "database is locked",
2023-02-13T09:00:42.969235355Z )
2023-02-13T09:00:42.969274197Z [2023-02-13 09:00:42.969][response][INFO] (invite_user) POST /admin/invite => 500 Internal Server Error
2023-02-13T09:00:45.211766491Z [2023-02-13 09:00:45.211][request][INFO] GET /admin/users/overview
2023-02-13T09:00:45.224925078Z [2023-02-13 09:00:45.224][response][INFO] (users_overview) GET /admin/users/overview => 200

```

I'm using your db.sqlite3 from this GitHub, after deployed I delete db.sqlite3 from fileshare folder and upload your db.sqlite3 version
I even do 'PRAGMA journal_mode=delete;' and replace the db.sqlite3 and still got the error "database is locked"

I'm not changing anything, just click Deploy to Azure button.
Can you help me with what is wrong or suggestions?

**Comment by abdulaziz-git** [comment]
Yes, I restarted couple of times, but still the same.

`2023-02-28T04:49:41.92361`  Connecting to the container 'vaultwarden'...

2023-02-28T04:49:41.94178  Successfully Connected to container: 'vaultwarden' [Revision: 'vaultwarden--p8ych2r', Replica: 'vaultwarden--p8ych2r-7486bf6d74-cn5t9']
2023-02-28T04:49:25.567056594Z /--------------------------------------------------------------------\
2023-02-28T04:49:25.567081653Z |                        Starting Vaultwarden                        |
2023-02-28T04:49:25.567085291Z |                           Version 1.27.0                           |
2023-02-28T04:49:25.567087916Z |--------------------------------------------------------------------|
2023-02-28T04:49:25.567090549Z | This is an *unofficial* Bitwarden implementation, DO NOT use the   |
2023-02-28T04:49:25.567093194Z | official channels to report bugs/features, regardless of client.   |
2023-02-28T04:49:25.567095504Z | Send usage/configuration questions or feature requests to:         |
2023-02-28T04:49:25.567097903Z |   https://vaultwarden.discourse.group/                             |
2023-02-28T04:49:25.567100304Z | Report suspected bugs/issues in the software itself at:            |
2023-02-28T04:49:25.567102807Z |   https://github.com/dani-garcia/vaultwarden/issues/new            |
2023-02-28T04:49:25.567105922Z \--------------------------------------------------------------------/
2023-02-28T04:49:25.567108425Z 
2023-02-28T04:49:25.567120827Z [INFO] No .env file found.
2023-02-28T04:49:25.567123615Z 
2023-02-28T04:49:25.978126131Z [2023-02-28 04:49:25.977][start][INFO] Rocket has launched from http://0.0.0.0:80

2023-02-28T04:49:47.670674777Z [2023-02-28 04:49:47.670][request][INFO] GET /admin/users/overview?

2023-02-28T04:49:47.705275173Z [2023-02-28 04:49:47.705][response][INFO] (users_overview) GET /admin/users/overview => 200 OK

2023-02-28T04:49:51.842382583Z [2023-02-28 04:49:51.842][vaultwarden::api::web][ERROR] Static file not found: bootstrap.css.map

2023-02-28T04:50:00.952946837Z [2023-02-28 04:50:00.952][request][INFO] POST /admin/invite/

2023-02-28T04:50:06.069715729Z [2023-02-28 04:50:06.069][error][ERROR] Error saving invitation.
2023-02-28T04:50:06.069759398Z [CAUSE] DatabaseError(
2023-02-28T04:50:06.069765730Z     Unknown,
2023-02-28T04:50:06.069770741Z     "database is locked",
2023-02-28T04:50:06.069775187Z )
2023-02-28T04:50:06.069785202Z [2023-02-28 04:50:06.069][response][INFO] (invite_user) POST /admin/invite => 500 Internal Server Error
`

**Comment by adamhnat** [comment]
it seems that there was error when replacing database with no WAL, can you please retry to copy file db.sqlite3 and make sure that in root of file share you have the following files:
![obraz](https://user-images.githubusercontent.com/15192744/221875847-80fa766e-ca7a-4f9d-b16a-331194b784cd.png)

Just redeployed and after replacing the database file - it works fine

**Comment by abdulaziz-git** [comment]
Hi @adamhnat,

There's no problem with the deployment, I can deploy it successfully
![Screenshot_7](https://user-images.githubusercontent.com/77265319/222340217-ddb9c757-4ec0-4b78-8a3e-cf362740bd54.png)

But the problem is when you try to invite a user from the admin panel, I get the following error:
![Screenshot_3](https://user-images.githubusercontent.com/77265319/222339306-03b220ed-4912-499c-8da6-ae1ea7d2ad51.png)

I also tried another way by registering a user and got the following error:
![Screenshot_4](https://user-images.githubusercontent.com/77265319/222339532-76a20fb0-d8c7-4df4-aef9-1a380646ae4a.png)

And this is the error from the log:
![Screenshot_5](https://user-images.githubusercontent.com/77265319/222340257-b3123193-4b50-4b98-940b-a8140b11f473.png)

I also check the ENABLE_DB_WAL is false
![Screenshot_6](https://user-images.githubusercontent.com/77265319/222340397-ae9afe22-ff97-4a52-b835-aea1668763d6.png)


Can you add a user to your deployment?
Is it a success?

**Comment by masyanru** [comment]
yeap, I have the same issue, just checked container logs 

2023-03-04T20:27:04.788725765Z [CAUSE] DatabaseError(
2023-03-04T20:27:04.788734865Z     Unknown,
2023-03-04T20:27:04.788741165Z     "database is locked",
2023-03-04T20:27:04.788746465Z )

**Comment by adamhnat** [comment]
I was able to recreate issue - seems that .sqlite3 database placed on SMB share isn't best idea - and after looking for it - it seems that's very common issue when putting .sqlite3 database on smbshare.
On free time I will rewrite scripts to use postgresql database intead of sqlite

**Comment by adamhnat** [comment]
issue resolved with commit [632638126555cf68819bde21143026a39eed6cbb](https://github.com/adamhnat/vaultwarden-azure/commit/632638126555cf68819bde21143026a39eed6cbb)

**Comment by abdulaziz-git** [comment]
@masyanru try deploying my pull request

[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fadamhnat%2Fvaultwarden-azure%2Fmaster%2Fmain.json)
